extensions/Khronos/KHR_materials_cmnConstant/README.md

```diff
@@ -0,0 +1,119 @@
+should be ignored.
+
+```json
+{
+    "materials": [
+        {
+            "name": "MyUnlitMaterial",
+            "pbrMetallicRoughness": {
+                "baseColorFactor": [ 0.5, 0.8, 0.0, 1.0 ]
+            },
+            "extensions": {
+                "KHR_materials_cmnConstant": {}
+            }
+        }
+    ]
+}
+```
+
+### Definition
+
+The Constant material model describes a constantly shaded surface that is
+independent of lighting. The material is defined only by properties already
+present in the [glTF 2.0 material specification](https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#material).
+No new properties are added by this extension — it is effectively a boolean
+flag indicating use of an unlit shading model. Additional properties on the
+extension object are allowed, but may lead to undefined behaviour in conforming
+viewers.
+
+Color is calculated as:
+
+```
+color = <emissiveTerm> + <baseColorTerm> * <occlusionTerm> * <aL>
+```
+
+Term definitions:
+
+* **emissiveTerm** – `emissiveFactor` * `emissiveTexture`
+    *  [*From core glTF material specification.*](https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#additional-maps)
+* **occlusionTerm** – `occlusionTexture.strength` * `occlusionTexture`
+    *  [*From core glTF material specification.*](https://github.com/KhronosGroup/glTF/tree/master/specification/2.0#additional-maps)
+* **baseColorTerm** – `baseColorFactor` * `baseColorTexture` * `vertexColor`
+    *  [*From core glTF material specification.*](https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/README.md#metallic-roughness-material)
+* **aL** – Indirect light
+    * *Constant sum of all ambient light contributions from the scene.*
+
+### Example
+
+The following example defines an unlit material with a constant red color.
+
+```json
+"materials": [
+    {
+        "name": "red_constant_material",
+        "pbrMetallicRoughness": {
+            "baseColorFactor": [ 1.0, 0.0, 0.0, 1.0 ]
+        },
+        "extensions": {
+            "KHR_materials_cmnConstant": {}
+        }
+    }
+]
+```
+
+### Extension compatibility and fallback materials
+
+When possible, authoring tools should provide a fallback material definition
+and mark the `KHR_materials_cmnConstant` extension as optional. Models
+including the extension optionally will still render in all clients that
+support the core glTF 2.0 specification, falling back to a PBR metal/rough
+material. In clients that do not support the extension, direct lighting will
+be applied and performance may be degraded.
+
+Defining a fallback to the default PBR shading model is achieved by setting
+appropriate parameters on the default material.
+
+```json
+"materials": [
+    {
+        "name": "unlit_with_fallback",
+        "pbrMetallicRoughness": {
+            "baseColorFactor": [ 1.0, 1.0, 1.0, 1.0 ],
+            "baseColorTexture": { "index": 0 },
+            "metallicFactor": 0.0,
```

Comment:
Now that I think about this more. I think we should note that the fallback should not have `metallicRoughnessTexture` and `metallicFactor` should be 0 and `roughnessFactor` should be 1. Same for `normalTexture` depending on what we decide for that.

Reply:
Ok. Any thoughts on whether that should be normative, or simply an implementation note for recommended fallback behavior?

Reply:
I think since this is already in a fallback section it doesn't make sense to be an implementation note. Normative is my vote.

Reply:
Following this through, does this look right?

- metallicRoughnessTexture CANNOT be present
- metallicFactor MUST be 0
- roughnessFactor MUST be 1

- occlusionTexture CANNOT be present
- emissiveTexture CANNOT be present
- emissiveFactor MUST be 0
- normalTexture CANNOT be present
- NORMAL and TANGENT attributes ...
  - cannot be present?
  - may be present, but must be ignored during lighting calculations?

Inclined to say NORMAL attribute should not be disallowed, but ignored in shading. Arguably it could be used for things other than shading e.g. for aligning a reticle against the surface. Same is arguably true of `normalTexture` though...

Reply:
> Inclined to say NORMAL attribute should not be disallowed, but ignored in shading. Arguably it could be used for things other than shading e.g. for aligning a reticle against the surface. Same is arguably true of `normalTexture` though...

+1 for the NORMAL attribute. I don't think the same is true for the `normalTexture`. It would be weird if there is no visible changes for the `normalTexture` but something else (like a reticle or physics) did something with it.

Reply:
Why should we dictate what the factors should be for the fallback case?  If I want a metallic or nonmetallic fallback, or shiny or rough, or whatever, I should be able to define that, right?

If you handle the extension you ignore everything that's not part of the lighting calculation above, and if you don't (or don't wish to) use the extension, you can use the fallback values.

Reply:
Also, huge +1 to renaming this KHR_materials_unlit

I never even considered that constant would mean anything but unlit, so maybe we should be explicit in the name and remove any references to 'constant'.

Reply:
Perhaps this should be an implementation note then